packages/react-client/src/Client.ts

```diff
@@ -559,55 +546,109 @@ export class Client<PeerMetadata, TrackMetadata> extends (EventEmitter as {
       this.emit("bandwidthEstimationChanged", estimation, this);
     });
 
-    this.deviceManager.on("deviceDisabled", (event) => {
+    this.videoDeviceManager.on("deviceDisabled", (state) => {
+      this.stateToSnapshot();
+
+      this.emit("deviceDisabled", { ...state, trackType: "video", mediaDeviceType: "userMedia" }, this);
+    });
+
+    this.videoDeviceManager.on("deviceEnabled", (state) => {
+      this.stateToSnapshot();
+
+      this.emit("deviceEnabled", { ...state, trackType: "video", mediaDeviceType: "userMedia" }, this);
+    });
+
+    this.videoDeviceManager.on("managerInitialized", () => {
+      this.stateToSnapshot();
+
+      this.emit("managerInitialized", { mediaDeviceType: "userMedia" }, this);
+    });
+
+    this.videoDeviceManager.on("managerStarted", (state) => {
+      this.stateToSnapshot();
+
+      this.emit("managerStarted", { ...state, trackType: "video", mediaDeviceType: "userMedia" }, this);
+    });
+
+    this.videoDeviceManager.on("deviceStopped", (state) => {
+      this.stateToSnapshot();
+
+      this.emit("deviceStopped", { ...state, trackType: "video", mediaDeviceType: "userMedia" }, this);
+    });
+
+    this.videoDeviceManager.on("deviceReady", ({ stream }) => {
+      this.stateToSnapshot();
+
+      this.emit("deviceReady", { trackType: "video", stream, mediaDeviceType: "userMedia" }, this);
+    });
+
+    this.videoDeviceManager.on("devicesStarted", (event) => {
+      this.stateToSnapshot();
+
+      this.emit("devicesStarted", { ...event, trackType: "video", mediaDeviceType: "userMedia" }, this);
+    });
+
+    this.videoDeviceManager.on("devicesReady", (event) => {
+      this.stateToSnapshot();
+
+      this.emit("devicesReady", { ...event, trackType: "video", mediaDeviceType: "userMedia" }, this);
+    });
+
+    this.videoDeviceManager.on("error", (event) => {
+      this.stateToSnapshot();
+
+      this.emit("error", event, this);
+    });
+
+    this.audioDeviceManager.on("deviceDisabled", (state) => {
       this.stateToSnapshot();
 
-      this.emit("deviceDisabled", { ...event, mediaDeviceType: "userMedia" }, this);
+      this.emit("deviceDisabled", { ...state, trackType: "audio", mediaDeviceType: "userMedia" }, this);
     });
 
-    this.deviceManager.on("deviceEnabled", (event) => {
+    this.audioDeviceManager.on("deviceEnabled", (state) => {
       this.stateToSnapshot();
 
-      this.emit("deviceEnabled", { ...event, mediaDeviceType: "userMedia" }, this);
+      this.emit("deviceEnabled", { ...state, trackType: "audio", mediaDeviceType: "userMedia" }, this);
     });
 
-    this.deviceManager.on("managerInitialized", (event) => {
+    this.audioDeviceManager.on("managerInitialized", () => {
       this.stateToSnapshot();
 
-      this.emit("managerInitialized", { ...event, mediaDeviceType: "userMedia" }, this);
+      this.emit("managerInitialized", { mediaDeviceType: "userMedia" }, this);
```

Comment:
Only the `userMedia` manager has an `init` method and can be initialized, so I think this` { mediaDeviceType: "userMedia" }` isn't very useful